src/Stryker.Core/Stryker.Core/Options/Inputs/BaselineProviderInput.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+using System.Linq;
+using Stryker.Core.Baseline.Providers;
+using Stryker.Core.Exceptions;
+using Stryker.Core.Reporters;
+
+namespace Stryker.Core.Options.Inputs
+{
+    public class BaselineProviderInput : InputDefinition<string, BaselineProvider>
+    {
+        public override string Default => "disk";
+
+        protected override string Description => "Choose a storage location for dashboard compare. Set to Dashboard provider when the dashboard reporter is turned on.";
+        protected override string HelpOptions => FormatEnumHelpOptions();
+
+        public BaselineProvider Validate(IEnumerable<Reporter> reporters)
+        {
+            if (SuppliedInput is null && reporters.Contains(Reporter.Dashboard))
+            {
+                return BaselineProvider.Dashboard;
+            }
+            else if (SuppliedInput is { })
```

Comment:
I probably missed something new in C#... this is comparing to what?

Reply:
This is a pattern match against 'not null object'. We can now replace this with the 'is not null' pattern but this wasn't available in c# 8. I started using it because pattern matching is safer than != due to operator overloading and '!(x is null)' being ugly :p

I believe mads torgedsen showed me this concept so don't blame me! 😂